app/controllers/course/survey/responses_controller.rb

```diff
@@ -27,7 +28,11 @@ def show
     render 'course/survey/surveys/index'
   end
 
+  # This action returns data that will be displayed the show page. Since the updates made
+  # are not specified by the user, we only ensure that user has permission to
+  # :read the response.
   def edit
+    authorize!(:read, @response)
```

Comment:
looks like we need to define a new permission `unsubmit`, but filter out the answer attributes in `update` if the user cannot edit the survey. 

I think edit is fine since staff need unsubmit. but you can also use `show` action for staff, and don't create the answers there.